lib/gamnit/depth/more_materials.nit

```diff
@@ -237,21 +244,20 @@ class TexturedMaterial
 		program.coord.array_enabled = true
 		program.coord.array(mesh.vertices, 3)
 
-		program.rotation.uniform new Matrix.gamnit_euler_rotation(actor.pitch, actor.yaw, actor.roll)
+		program.rotation = new Matrix.gamnit_euler_rotation(actor.pitch, actor.yaw, actor.roll)
 
-		var a = actor.alpha
-		program.ambient_color.uniform(ambient_color[0]*a, ambient_color[1]*a,
-		                              ambient_color[2]*a, ambient_color[3]*a)
-		program.diffuse_color.uniform(diffuse_color[0]*a, diffuse_color[1]*a,
-		                              diffuse_color[2]*a, diffuse_color[3]*a)
-		program.specular_color.uniform(specular_color[0]*a, specular_color[1]*a,
-		                               specular_color[2]*a, specular_color[3]*a)
+		program.ambient_color.uniform(ambient_color[0], ambient_color[1],
```

Comment:
I guess there is a really good reason why `TexturedMaterial` and `SmoothMaterial` have the same code ?

Reply:
At first, I found it clearer to keep them distinct but not anymore. My optimizations replace this method and in favor of a new one without this duplication.